networkx/algorithms/similarity.py

```diff
@@ -1386,3 +1393,249 @@ def simrank_similarity_numpy(
     if source is not None:
         return newsim[source]
     return newsim
+
+
+def n_choose_k(n, k):
+    """Pure Python implementation of the binomial coefficient
+
+    The general equation is n! / (k! * (n - k)!). The below
+    implementation is a more efficient version.
+
+    Parameters
+    ----------
+    n : int
+        Set of ``n`` elements
+    k : int
+        Unordered chosen subset of length ``k``
+
+    Returns
+    -------
+    binomial_coeff : int
+        The number of ways (disregarding order) that k objects
+        can be chosen from among n objects.
+
+    Examples
+    --------
+        >>> import networkx as nx
+        >>> nx.n_choose_k(5, 2)
+        10
+        >>> nx.n_choose_k(5, 4)
+        5
+        >>> nx.n_choose_k(100, 100)
+        1
+
+    """
+    if k > n:
+        return 0
+    if n == k:
+        return 1
+    elif k < n - k:
+        return reduce(mul, range(n - k + 1, n + 1)) // math.factorial(k)
+    else:
+        return reduce(mul, range(k + 1, n + 1)) // math.factorial(n - k)
+
+
+def panther_similarity(G, v, k=5, path_length=5, c=0.5, delta=0.1, eps=None):
+    """Returns the Panther similarity of nodes in the graph ``G`` to node ``v``.
+    Panther is a similarity metric that says "two objects are considered
+    to be similar if they frequently appear on the same paths." [1]_.
+
+    Parameters
+    ----------
+    G : NetworkX graph
+        A NetworkX graph
+    v : node
+        Source node for whom to find the top ``k`` similar other nodes
+    k : int
+        The number of most similar nodes to return
+    path_length : int
+        How long the randomly generated paths should be (``T`` in [1]_)
+    c : float
+        A universal positive constant. Defaults to 0.5 per [1]_
```

Comment:
c: float (default = 0.5)
    A universal positive constant used to scale the number of sample random paths to generate.

Reply:
Addressed in 221da98.